apps/web/src/dialogs/command-palette/command-palette-dialog.tsx

```diff
@@ -0,0 +1,595 @@
+                      <Button
+                        title="Remove from recent"
+                        onClick={(e) => {
+                          e.stopPropagation();
+                          removeRecentCommand(command.id);
+                          setCommands((commands) =>
+                            commands.filter((c) => c.id !== command.id)
+                          );
+                        }}
+                        variant="icon"
+                        sx={{
+                          bg:
+                            selected.x === 1 && index === selected.y
+                              ? "hover"
+                              : "transparent",
+                          p: 1,
+                          mr: 1,
+                          ":hover:not(:disabled):not(:active)": {
+                            bg: "hover"
+                          }
+                        }}
+                      >
+                        <Cross size={14} />
+                      </Button>
+                    )}
+                  </Flex>
+                );
+              }}
+            />
+          </Box>
+        </Flex>
+        <Flex
+          sx={{ flexDirection: "row", bg: "hover", justifyContent: "center" }}
+        >
+          <Text variant="subBody" sx={{ m: 0.5 }}>
+            <kbd>{">"}</kbd> for command mode · remove <kbd>{">"}</kbd> for
+            search mode · <kbd>⏎</kbd> to select · <kbd>↑</kbd>
+            <kbd>↓</kbd> to navigate
+          </Text>
+        </Flex>
+      </Dialog>
+    );
+  }
+);
+
+const arrowNavigate: Record<
+  "down" | "up" | "right" | "left",
+  (
+    current: Record<"x" | "y", number>,
+    commands: Command[]
+  ) => Record<"x" | "y", number>
+> = {
+  down: (current, commands) => {
+    const currentCommand = commands[current.y];
+    const nextIndex = (current.y + 1) % commands.length;
+    const nextCommand = commands[nextIndex];
+    if (currentCommand.group === "recent" && nextCommand.group === "recent") {
+      return { x: current.x, y: nextIndex };
+    }
+    return { x: 0, y: nextIndex };
+  },
+  up: (current, commands) => {
+    const currentCommand = commands[current.y];
+    const nextIndex = (current.y - 1 + commands.length) % commands.length;
+    const nextCommand = commands[nextIndex];
+    if (currentCommand.group === "recent" && nextCommand.group === "recent") {
+      return { x: current.x, y: nextIndex };
+    }
+    return { x: 0, y: nextIndex };
+  },
+  right: (current, commands) => {
+    const currentCommand = commands[current.y];
+    if (currentCommand.group !== "recent") return current;
+    const nextIndex = (current.x + 1) % 2;
+    return { x: nextIndex, y: current.y };
+  },
+  left: (current, commands) => {
+    const currentCommand = commands[current.y];
+    if (currentCommand.group !== "recent") return current;
+    const nextIndex = (current.x - 1 + 2) % 2;
+    return { x: nextIndex, y: current.y };
+  }
+};
```

Comment:
This is super confusing. Let's split these into their own functions that clearly describe what exactly they do instead of something like `arrowUp`, `arrowDown` etc. Function names should clearly explain what the function does instead of describing where its called or how its invoked.

Reply:
I've renamed them to `moveSelectionRight` etc

Let me know if that works